### ISSUE TITLE
call to reactor.spawnProcess in docs should be passed protocol instance

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3237]
|<img alt="terrycojones's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| terrycojones reported|
|-|-|
|Trac ID|trac#3237|
|Type|defect|
|Created|2008-05-15 21:04:47Z|

The first example of calling `reactor.spawnProcess` on this page
http://twistedmatrix.com/projects/core/documentation/howto/process.html should pass `mypp` not `processProtocol` as the first argument.

<details><summary>Searchable metadata</summary>

```
trac-id__3237 3237
type__defect defect
reporter__terrycojones terrycojones
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1210885487000000 1210885487000000
changetime__1210952875000000 1210952875000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

```
Index: doc/core/howto/process.xhtml
===================================================================
--- doc/core/howto/process.xhtml        (revision 23611)
+++ doc/core/howto/process.xhtml        (working copy)
@@ -36,7 +36,7 @@
 <pre class="python">
 from twisted.internet import reactor

-mypp = MyProcessProtocol()
+processProtocol = MyProcessProtocol()
 reactor.spawnProcess(processProtocol, executable, args=[program, arg1, arg2],
                      env={'HOME': os.environ['HOME']}, path,
                      uid, gid, usePTY, childFDs)
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Hum... every other arguments are undefined, so I'm sure this clarifies some many things. But well, please apply.

**Comment by twisted-trac** [comment]
|<img alt="terrycojones's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|terrycojones commented|
|-|-|

I think it's fine that args aren't defined - they're clearly examples. But the way this snippet stood was misleading. A variable was declared on the first line and then not used on the second. The change also makes it easier to infer that the first arg to spawnProcess should be an instance of a protocol.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [23635]) Apply patch to use consistent names in process example

Author: exarkun
Reviewer: therve
Fixes: [#3237](https://github.com/twisted/twisted/issues/3237)

An example in the process howto created one local variable and passed another undefined
variable to reactor.spawnProcess.  Change it to pass the variable it defines.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
